### PR TITLE
Add support for terraform-plugin-framework migrated resources.

### PR DESCRIPTION
**Pull request by danielmkn**
Add utilfw to use in a new resources migrated to the framework.

### CODE REVIEW

File: utilfw/utilfw.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// ValueToSetType ensures we have a types.Set literal
+func ValueToSetType(v attr.Value) types.Set {
+	if vb, ok := v.(types.Set); ok {
+		return vb
+	} else if vb, ok := v.(*types.Set); ok {
```
Comment (chb0github) [inline utilfw/utilfw.go:14]: If you have a `return` from an `if` you don't need an `else`, you just need 3 `if-return`

File: utilfw/utilfw.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		return *vb
+	} else {
```
Comment (chb0github) [inline utilfw/utilfw.go:16]: You can literally remove this `else`

**Review by chb0github — COMMENTED** [review]
You can simplify your flow control

File: utilfw/utilfw.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package utilfw
```
Comment (alexhung) [inline utilfw/utilfw.go:1]: Not sure we need another directory for this. Why not put this in `/util/framework.go`? Of if you like short name, `util/fw.go`?

**Review by alexhung — CHANGES_REQUESTED** [review]
What @chb0github said, as well as directory location for the new file.

File: utilfw/utilfw.go (hunk 3 above)
Comment (danielmkn) [reply utilfw/utilfw.go:1]: We need another package name. Can't have more than one package in the folder, if it's not `<packagename>_test`.

File: utilfw/utilfw.go (hunk 1 above)
Comment (danielmkn) [reply utilfw/utilfw.go:14]: yes, agree, thanks!

File: utilfw/utilfw.go (hunk 3 above)
Comment (danielmkn) [reply utilfw/utilfw.go:1]: Changed to `/util/sdk`  and `/util/fw`.

**Review by alexhung — DISMISSED** [review]
Ideally, `util/sdk/util.go` should be `util/sdk/sdk.go` (likewise with `fw.go`) but this is fine for now.

**Comment by danielmkn** [comment]
Renamed, let's have it right.

**Pull request merged**